python/pyarrow/io.pxi

```diff
@@ -989,9 +1035,67 @@ cdef class BufferReader(NativeFile):
         else:
             self.buffer = py_buffer(obj)
 
-        self.rd_file.reset(new CBufferReader(self.buffer.buffer))
+        self.set_random_access_file(shared_ptr[RandomAccessFile](
+            new CBufferReader(self.buffer.buffer)))
         self.is_readable = True
-        self.closed = False
+
+
+cdef class CompressedInputStream(NativeFile):
+    """
+    An input stream wrapper which decompresses data on the fly.
+
+    Parameters
+    ----------
+    stream : pa.NativeFile
+    compression : str
+        The compression type (e.g. "gzip" or "bz2")
```

Comment:
is lz4 supported or just these two?

Reply:
lz4 is supported too (see roundtrip tests).

Reply:
Ok, should list the supported compressors here or direct the user where to find them